core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -780,46 +774,36 @@ class HttpGoogleServicesDAO(val clientSecrets: GoogleClientSecrets,
    */
   override def setBillingAccountName(googleProjectId: GoogleProjectId,
                                      billingAccountName: RawlsBillingAccountName,
-                                     span: Span = null
+                                     tracingContext: RawlsTracingContext
   ): Future[ProjectBillingInfo] = {
     // Since this method should only be called with a non-empty Billing Account Name, then Rawls should also make sure
     // that Billing is enabled for the Google Project, otherwise users' projects could get "stuck" in a Disabled Billing
     // state with no way to reenable because they do not have permissions to do this directly on the Google Project.
     val newProjectBillingInfo =
       new ProjectBillingInfo().setBillingAccountName(billingAccountName.value).setBillingEnabled(true)
-    updateBillingInfo(googleProjectId, newProjectBillingInfo, span)
+    updateBillingInfo(googleProjectId, newProjectBillingInfo, tracingContext)
   }
 
-  override def disableBillingOnGoogleProject(googleProjectId: GoogleProjectId): Future[ProjectBillingInfo] = {
+  override def disableBillingOnGoogleProject(googleProjectId: GoogleProjectId, tracingContext: RawlsTracingContext): Future[ProjectBillingInfo] = {
     val newProjectBillingInfo = new ProjectBillingInfo().setBillingEnabled(false)
-    updateBillingInfo(googleProjectId, newProjectBillingInfo)
+    updateBillingInfo(googleProjectId, newProjectBillingInfo, tracingContext)
   }
 
   private def updateBillingInfo(googleProjectId: GoogleProjectId,
                                 projectBillingInfo: ProjectBillingInfo,
-                                parentSpan: Span = null
+                                tracingContext: RawlsTracingContext
   ): Future[ProjectBillingInfo] = {
     implicit val service = GoogleInstrumentedService.Billing
     val billingSvcCred = getBillingServiceAccountCredential
     val cloudBillingProjectsApi = getCloudBillingManager(billingSvcCred).projects()
 
-    traceWithParent("cloudBillingProjectsApi.updateBillingInfo", parentSpan) { s =>
+    traceFutureWithParent("cloudBillingProjectsApi.updateBillingInfo", tracingContext) { childContext =>
       val updater = cloudBillingProjectsApi.updateBillingInfo(s"projects/${googleProjectId.value}", projectBillingInfo)
       retryWithRecoverWhen500orGoogleError { () =>
         blocking {
-          val span = startSpanWithParent("executeGoogleRequest", s)
-          span.putAttribute("googleProjectId", AttributeValue.stringAttributeValue(googleProjectId.value))
-          span.putAttribute("billingAccount",
-                            AttributeValue.stringAttributeValue(Option(projectBillingInfo.getBillingAccountName) match {
-                              case Some(value) => value
-                              case None        => ""
-                            })
-          )
-
-          try
+          traceNakedWithParent("executeGoogleRequest", childContext) { _ =>
```

Comment:
does this lose the `googleProjectId` and `billingAccount` attributes on the span?